### PR TITLE
Refactor and support Android 10

### PR DESCRIPTION
**Pull request by sledges**

### CODE REVIEW

File: helpers/copy_tree.sh [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+# Remove generic unused directories and files
+source "$(get_closest helpers/remove-unused-"$TREE".sh)"
+
+# General patches
+echo "Apply general patches:"
+source droid-system-device/helpers/general-patches.sh
+
+# Move build.prop to proper place if spec file has multiple_rpms definition
+if grep -q "%define multiple_rpms 1" "$modify_spec"; then
+    if [ "$TREE" == "system" ]; then
```
Comment (Thaodan) [inline helpers/copy_tree.sh:89]: Use = instead of == here or use [[]].

File: helpers/copy_tree.sh [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                exit 1
+            fi
+            mkdir -p "$DEVICE"/system
+            mv "$TREE_SPARSE"/build.prop "$DEVICE"/system
+            if [ -f "$TREE_SPARSE"/etc/prop.default ]; then
+                mkdir -p "$DEVICE"/system/etc
+                mv "$TREE_SPARSE"/etc/prop.default "$DEVICE"/system/etc/prop.default
+            fi
+        fi
+    elif [ "$TREE" == "vendor" ]; then
```
Comment (Thaodan) [inline helpers/copy_tree.sh:106]: Same here.

File: helpers/fixup-permissions-system.sh [hunk 3]
```diff
@@ -2,17 +2,13 @@
...
 # Add read permission for some binaries under system mount
 
-if [ "$SYSTEM_MOUNT" == "" ]; then
-    SYSTEM_MOUNT=$VENDOR_MOUNT
-fi
-
-if [ "$SYSTEM_MOUNT" == "" ]; then
-    echo "SYSTEM_MOUNT and VENDOR_MOUNT are unset!"
+if [ "$TREE_MOUNT" == "" ]; then
+    echo "TREE_MOUNT is unset!"
```
Comment (Thaodan) [inline helpers/fixup-permissions-system.sh:6]: Same here.

File: helpers/fixup-permissions-vendor.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Add read permission for some binaries under system mount
+
+if [ "$TREE_MOUNT" == "" ]; then
+    echo "TREE_MOUNT is unset!"
```
Comment (Thaodan) [inline helpers/fixup-permissions-vendor.sh:6]: Same here.

**Review by Thaodan — CHANGES_REQUESTED** [review]

File: helpers/fixup-permissions-vendor.sh (hunk 4 above)
Comment (sledges) [reply helpers/fixup-permissions-vendor.sh:6]: Fixed all bashisms (even if top of scripts reads #!/bin/bash, it can ease portability)

**Review by mkosola — APPROVED** [review]

**Pull request merged**